tests/tape/interfaces/test_qnode_jax.py

```diff
@@ -0,0 +1,205 @@
+		qml.RZ(weights[1], wires=1)
+		return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+	weights = jnp.array([0.1, 0.2])
+	val = circuit(weights)
+	assert "DeviceArray" in val.__repr__()
+
+def test_to_jax():
+	dev = qml.device("default.mixed", wires=2) 
+
+	@qml.qnode(dev, interface="autograd")
+	def circuit(weights):
+		qml.RX(weights[0], wires=0)
+		qml.RZ(weights[1], wires=1)
+		return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+	circuit.to_jax()
+	weights = jnp.array([0.1, 0.2])
+	val = circuit(weights)
+	assert "DeviceArray" in val.__repr__()
+
+
+def test_simple_jacobian():
+	dev = qml.device("default.mixed", wires=2) # A non-JAX device.
+
+	@qml.qnode(dev, interface="jax", diff_method="parameter-shift")
+	def circuit(weights):
+		qml.RX(weights[0], wires=0)
+		qml.RZ(weights[1], wires=1)
+		return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+	weights = jnp.array([0.1, 0.2])
+	val = jax.jacrev(circuit)(weights)
+	assert "DeviceArray" in val.__repr__()
+
+def test_simple_grad():
+	dev = qml.device("default.mixed", wires=2) # A non-JAX device.
+	@qml.qnode(dev, interface="jax", diff_method="parameter-shift")
+	def circuit(weights):
+		qml.RX(weights[0], wires=0)
+		qml.RZ(weights[1], wires=1)
+		return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+	weights = jnp.array([0.1, 0.2])
+	val = jax.grad(circuit)(weights)
+	assert "DeviceArray" in val.__repr__()
+
+@pytest.mark.parametrize("diff_method", ['parameter-shift', 'finite-diff'])
+def test_differentiable_expand(diff_method):
+    """Test that operation and nested tapes expansion
+    is differentiable"""
+    class U3(qml.U3):
+        def expand(self):
+            theta, phi, lam = self.data
+            wires = self.wires
+
+            with JacobianTape() as tape:
+                qml.Rot(lam, theta, -lam, wires=wires)
+                qml.PhaseShift(phi + lam, wires=wires)
+
+            return tape
+
+    dev = qml.device("default.mixed", wires=1)
+    a = jnp.array(0.1)
+    p = jnp.array([0.1, 0.2, 0.3])
+
+    @qnode(dev, diff_method=diff_method, interface="jax")
+    def circuit(a, p):
+        qml.RX(a, wires=0)
+        U3(p[0], p[1], p[2], wires=0)
+        return qml.expval(qml.PauliX(0))
+
+    res = circuit(a, p)
+
+    expected = np.cos(a) * np.cos(p[1]) * np.sin(p[0]) + np.sin(a) * (
+        np.cos(p[2]) * np.sin(p[1]) + np.cos(p[0]) * np.cos(p[1]) * np.sin(p[2])
+    )
+    tol = 1e-5
+    assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    res = jax.grad(circuit, argnums=1)(a, p)
+    expected = np.array(
+        [
+            np.cos(p[1]) * (np.cos(a) * np.cos(p[0]) - np.sin(a) * np.sin(p[0]) * np.sin(p[2])),
+            np.cos(p[1]) * np.cos(p[2]) * np.sin(a)
+            - np.sin(p[1])
+            * (np.cos(a) * np.sin(p[0]) + np.cos(p[0]) * np.sin(a) * np.sin(p[2])),
+            np.sin(a)
+            * (np.cos(p[0]) * np.cos(p[1]) * np.cos(p[2]) - np.sin(p[1]) * np.sin(p[2])),
+        ]
+    )
+    assert np.allclose(res, expected, atol=tol, rtol=0)

```

Comment:
Maybe this is copied from other interfaces, but if the test checks "unnesting" would one not rather check a tape queue here?

Reply:
I don't understand what you mean.

Reply:
Oh I think I understand now on my side...you are not testing the expansion itself in this sense...all good.